### PR TITLE
Fix wrong SDK being reported when sentry_logging is used

### LINKED ISSUE #717: Fix sdk reporting of `sentry_logging`
Looks like this should not be set https://github.com/getsentry/sentry-dart/blob/main/logging/lib/src/logging_integration.dart#L49-L50
Since its an Integration rather than a top-level package, adding to the package list https://github.com/getsentry/sentry-dart/blob/main/logging/lib/src/logging_integration.dart#L54
Adding to the integration list https://github.com/getsentry/sentry-dart/blob/main/logging/lib/src/logging_integration.dart#L38 and setting the logger https://github.com/getsentry/sentry-dart/blob/main/logging/lib/src/extension.dart#L27 should have been enough.
Would you like to submit a PR @kuhnroyal ?

_Originally posted by @marandaneto in https://github.com/getsentry/sentry-dart/pull/631#discussion_r792709472_

### PR DESCRIPTION
**Pull request by kuhnroyal**
## :scroll: Description
Fixes #717 

## :green_heart: How did you test it?
I checked the Sentry UI, the package doesn't seem to be reported anywhere tho.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Unreleased
+* Fix: `sentry_logging` incorrectly setting SDK name
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion

* Fix: `sentry_logging` incorrectly setting SDK name (#725)
```

**Review by marandaneto — APPROVED** [review]
thanks @kuhnroyal

**Review by brustolin — COMMENTED** [review]
LGTM

**Comment by marandaneto** [comment]
@kuhnroyal `package-analysis` counted -10 points using `pana`, mind checking it out?

**Comment by kuhnroyal** [comment]
@marandaneto Seems unrelated:
```
### [~] 20/30 points: code has no errors, warnings, lints, or formatting issues\n\n<details>\n<summary>\nINFO: Prefer typing uninitialized variables and fields.\n</summary>\n\n`lib/src/sentry_exception_factory.dart:19:9`\n\n```\n   ╷\n19 │     var mechanism;\n   │         ^^^^^^^^^\n   ╵\n```\n\nTo reproduce make sure you are using the [lints_core](pub.dev/packages/lints) and run `dart analyze lib/src/sentry_exception_factory.dart`\n</details>
```

**Comment by marandaneto** [comment]
> sentry_exception_factory

Thanks, `dart` package still uses `pedantic`, but this one is related to the `lints` package.
Time to address https://github.com/getsentry/sentry-dart/issues/507 :D

**Pull request merged**